Harbor can't handle the repository name contain "/"

**If you are reporting a problem, please make sure the following information are provided:**
1)Version of docker 1.12.3
2)Config files of harbor, just use http not https
3)Log files, you can get them by package the /var/log/harbor/ 
Dec  6 00:16:46 172.18.0.1 ui[26532]: 2016/13/06 05:16:46 #033[1;34m[router.go:829][D] | POST       | /service/notifications                   | 601.872µs        | match      | /service/notifications                   | #033[0m
Dec  6 00:16:46 172.18.0.1 ui[26532]: 2016-12-06T05:16:46Z [DEBUG] [notification.go:96]: Increase the repository replication1/vmware/harbor-db pull count.
Dec  6 00:16:46 172.18.0.1 ui[26532]: 2016-12-06T05:16:46Z [ERROR] [accesslog.go:154]: error in AccessLog: Error 1048: Column 'project_id' cannot be null
Dec  6 00:16:46 172.18.0.1 ui[26532]: 2016-12-06T05:16:46Z [ERROR] [accesslog.go:155]: sql command is: admin, replication1/vmware, replication1/vmware/harbor-db, dev, pull
Dec  6 00:16:46 172.18.0.1 ui[26532]: 2016-12-06T05:16:46Z [ERROR] [notification.go:73]: failed to add access log: Error 1048: Column 'project_id' cannot be null

**Analysis:** Harbor receive the notification after push repository registry complete, but if the more than two "/" in the repository name, harbor couldn't find the right project name, but this repository already exist in the Harbor's registry, for example: 

10.1.1.143/replication1/vmware/harbor-log
Harbor's ip is "1.1.1.143 "
project name is "replication1 "
but Harbor think project name is "replication1/vmware"

**Fix :**  "src/common/utils/utils.go :49"
func ParseRepository(repository string) (project, rest string) {
        repository = strings.TrimLeft(repository, "/")
        repository = strings.TrimRight(repository, "/")
        if !strings.ContainsRune(repository, '/') {
                rest = repository
                return
        }
        //index := strings.LastIndex(repository, "/")             //replace this line with following line
        index := strings.Index(repository, "/")
        project = repository[0:index]
        rest = repository[index+1:]
        return
}

--- a/src/common/utils/utils.go
+++ b/src/common/utils/utils.go
@@ -53,7 +53,7 @@ func ParseRepository(repository string) (project, rest string) {
 		rest = repository
 		return
 	}
-	index := strings.LastIndex(repository, "/")
+	index := strings.Index(repository, "/")
 	project = repository[0:index]
 	rest = repository[index+1:]
 	return

--- a/src/common/utils/utils_test.go
+++ b/src/common/utils/utils_test.go
@@ -65,10 +65,10 @@ func TestParseRepository(t *testing.T) {
 
 	repository = "library/test/ubuntu"
 	project, rest = ParseRepository(repository)
-	if project != "library/test" {
+	if project != "library" {
 		t.Errorf("unexpected project: %s != %s", project, "library/test")
 	}
-	if rest != "ubuntu" {
+	if rest != "test/ubuntu" {
 		t.Errorf("unexpected rest: %s != %s", rest, "ubuntu")
 	}
 

--- a/src/ui/api/utils.go
+++ b/src/ui/api/utils.go
@@ -151,7 +151,7 @@ func TriggerReplication(policyID int64, repository string,
 func GetPoliciesByRepository(repository string) ([]*models.RepPolicy, error) {
 	repository = strings.TrimSpace(repository)
 	repository = strings.TrimRight(repository, "/")
-	projectName := repository[:strings.LastIndex(repository, "/")]
+	projectName, _ := utils.ParseRepository(repository)
 
 	project, err := dao.GetProjectByName(projectName)
 	if err != nil {

